chain/chain/src/chain.rs

```diff
@@ -358,6 +358,15 @@ pub struct BlockMissingChunks {
     pub missing_chunks: Vec<ShardChunkHeader>,
 }
 
+impl Debug for BlockMissingChunks {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BlockMissingChunks")
```

Comment:
FWIW my personal preference is to log out individual fields rather than debug representations of ADTs as that allows for more fine-grained with tooling that may eventually get properly set up.

So for instance instead of:

```rust
debug!(?missing_chunks);
// missing_chunks=BlockMissingChunks { prev_hash: ..., num_missing_chunks: 42 }
```

I go for

```rust
debug!(
    "missing_chunks.prev_hash" = ?missing_chunks.prev_hash,
    "missing_chunks.num_missing_chunks" = missing_chunks.num_missing_chunks
);
```

Of course that’s more code, but for types that aren’t printed out too often, it is well worth the effort in my books.

Reply:
But it doesn't work for vectors containing these types.